### PR TITLE
DDCE-5074 Dependency upgrades, README.md updates & cleanup

### PR DESCRIPTION
**Pull request by Taiwo-AB**

### CODE REVIEW

**Comment by Taiwo-AB** [comment]
Full test run -> https://performance.tools.staging.tax.service.gov.uk/job/advance-valuation-rulings-performance-tests/218/
Smoke test run -> https://performance.tools.staging.tax.service.gov.uk/job/advance-valuation-rulings-performance-tests/219/

- Dependency upgrades
- README.md updates
- General cleanup

File: scalastyle-config.xml [hunk 1]
```diff
@@ -9,8 +9,19 @@
     <check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
         <parameters>
             <parameter name="header"><![CDATA[/*
- * Copyright 2023 HM Revenue & Customs
+ * Copyright 2024 HM Revenue & Customs
```
Comment (EdCope) [inline scalastyle-config.xml:12]: Shouldn't we be using short format for perf repo? just comparing to some of our others

File: src/test/scala/uk/gov/hmrc/perftests/ars/ArsRequests.scala [hunk 2]
```diff
@@ -173,189 +136,172 @@ object ArsRequests extends ServicesConfiguration {
...
     getPage(
       "Name Of Short description of goods",
-      true,
-      s"$baseUrl/advance-valuation-ruling/$${draftId}/give-short-description-goods"
+      saveToken = true,
+      s"$arsUrl/advance-valuation-ruling/$${draftId}/give-short-description-goods"
     )
-  def enterNameofTheGoods(enterText: Boolean) = {
+
+  def enterNameofTheGoods(): HttpRequestBuilder = {
```
Comment (EdCope) [inline src/test/scala/uk/gov/hmrc/perftests/ars/ArsRequests.scala:201]: Rename to `enterNameOfTheGoods`

File: src/test/scala/uk/gov/hmrc/perftests/ars/ArsRequests.scala [hunk 3]
```diff
@@ -173,189 +136,172 @@ object ArsRequests extends ServicesConfiguration {
...
     getPage(
       "Found Legal challenges Page",
-      true,
-      s"$baseUrl/advance-valuation-ruling/$${draftId}/any-legal-challenges"
+      saveToken = true,
+      s"$arsUrl/advance-valuation-ruling/$${draftId}/any-legal-challenges"
     )
 
-  def submityesOrNolChallengePage(answer: Boolean) =
+  def submitYesOrNolChallengePage(answer: Boolean): HttpRequestBuilder =
```
Comment (EdCope) [inline src/test/scala/uk/gov/hmrc/perftests/ars/ArsRequests.scala:255]: Is this `NolChallenge` in def name for representing `NoLegalChallenge` ?

**Review by EdCope — CHANGES_REQUESTED** [review]

File: scalastyle-config.xml (hunk 1 above)
Comment (EdCope) [reply scalastyle-config.xml:12]: Repo is still public so leaving til change

**Review by EdCope — APPROVED** [review]

**Pull request merged**